release/models/nat/openconfig-nat.yang

```diff
@@ -0,0 +1,965 @@
+      description
+        "Ending port number for NAPT translations";
+    }
+
+    leaf port-block-size {
+      type uint16;
+      description
+        "Number of ports allocated per mapping";
+    }
+
+    leaf acl-set {
+      type leafref {
+        path "/oc-acl:acl/oc-acl:acl-sets/oc-acl:acl-set/" +
+             "oc-acl:config/oc-acl:name";
+      }
+      description
+        "Reference to ACL set that defines traffic
+        eligible for this pool";
+    }
+
+    leaf persistent {
+      type boolean;
+      description
+        "Enable persistent NAT mapping";
+    }
+
+    leaf max-sessions {
+      type uint32;
+      description
+        "Maximum number of concurrent sessions";
+    }
+
+    leaf address-only {
+      type boolean;
+      description
+        "Enable address-only translation without port translation";
+    }
+
+    leaf timeout {
+      type uint32;
+      units "seconds";
+      description
+        "Default NAT mapping timeout";
+    }
+
+    leaf tcp-timeout {
+      type uint32;
+      units "seconds";
+      description
+        "TCP-specific mapping timeout";
+    }
+
+    leaf udp-timeout {
+      type uint32;
+      units "seconds";
+      description
+        "UDP-specific mapping timeout";
+    }
+
+    leaf icmp-timeout {
+      type uint32;
+      units "seconds";
+      description
+        "ICMP-specific mapping timeout";
+    }
+
+    leaf overload {
+      type boolean;
+      description
+        "Enable port overloading (PAT) allowing multiple internal
+        addresses to share the same external address with
+        different ports";
+    }
+
+    leaf log-translations {
+      type boolean;
+      description
+        "Enable logging of NAT translations";
+    }
+
+    leaf max-translations-per-address {
+      type uint32;
+      description
+        "Maximum number of concurrent translations per external
+        address";
+    }
+  }
+
+  grouping nat-pool-state {
+    description
+      "Operational state data for NAT address pools";
+
+    container counters {
+      description
+        "Counters for NAT pool operations";
+
+      leaf allocated-addresses {
+        type uint32;
+        description
+          "Number of addresses currently allocated from this pool";
+      }
+
+      leaf available-addresses {
+        type uint32;
+        description
+          "Number of addresses available in this pool";
+      }
+
+      leaf active-sessions {
+        type uint32;
+        description
+          "Number of currently active sessions using this pool";
+      }
+
+      leaf acl-hit-count {
+        type yang:counter64;
+        description
+          "Number of packets matching the ACL set";
+      }
+    }
+  }
+
+  grouping nat-mapping-config {
+    description
+      "Configuration parameters for static NAT mappings";
+
+    leaf name {
+      type string;
+      description
+        "Unique name for the static mapping";
+    }
+
+    leaf internal-address {
```

Comment:
Since this model is focused on NAT44, should we update the model prefix from `nat` to `nat44`? It may help reduce confusion if in the future OC is updated to support other forms of NAT

Reply:
I'll add an `address-family` leaf and remove any IPv4 references, so the model covers NAT44 and NAT66